packages/roosterjs-editor-dom/lib/list/VList.ts

```diff
@@ -182,10 +182,18 @@ export default class VList {
         this.rootList.parentNode.replaceChild(placeholder, this.rootList);
 
         this.items.forEach(item => {
+            if (item.getNewListStart()) {
+                listStack.splice(1, listStack.length - 1);
+            }
             item.writeBack(listStack, this.rootList);
             const topList = listStack[1];
 
-            if (safeInstanceOf(topList, 'HTMLOListElement')) {
+            if (item.getNewListStart()) {
```

Comment:
See my comment above. Once you set the start number above, no need to add extra code here, existing code should be able to handle it.